docs/MAINTAINERS.adoc

```diff
@@ -0,0 +1,200 @@
+  It is **fine** to close invalid and inactive pull requests if there is no activity by a submitter or other contributors.
+* Enforcing a particular coding style.
+  Jenkins core has a complex codebase created by many contributors, and different files have different designs.
+  Our main goal is to firstly have the code readable by other contributors.
+* Make contributors to fix unrelated issues
+** Create follow-up issues instead, it is fine to reference them in comments so that the contributor might want to pick them up
+* Breakdown the commit history or squashing PRs
+** Not every contributor is a Git expert, do not request changes in the commit history unless it is necessary
+** Core maintainers can squash PRs during the merge.
+   If you feel this is important, use the `squash-merge-me` label
+** We want to keep pull requests atomic when possible (one feature / fix - separate commit),
+   but we can live without it if there is no need to backport changes to the stable baseline.
+
+== Merge process
+
+=== Common merge process
+
+**Step 1: Maintainer checklist**. Merge process can be initiated once a pull request matches the requirements:
+
+* Pull request is compliant with requirements to submitters (see the link:/.github/PULL_REQUEST_TEMPLATE.md[pull request template])
+* There are at least 2 approvals for the pull request and no outstanding requests for change
+* Conversations in the pull request are over OR it is explicit that a reviewer does not block the change (often indicated by line comments attached to an approving PR review, or by using the term "nit", from "nit-picking")
+* Changelog entries in the PR title and/or `Proposed changelog entries` are correct and reflect the current, final state of the PR
+* Proper changelog labels are set so that the changelog can be generated automatically
+* If the change needs administrators to be aware of it when upgrading, the `upgrade-guide-needed` label is present. This is usually the case when a data migration occurs, a feature has been removed, a significant behavior change is introduced (including when there is a way to opt out), or in general when we expect at least a large minority of admins to benefit from knowing about the change, e.g. to apply a new option.
+  and there is a `Proposed upgrade guidelines` section in the PR title.
+  (link:https://github.com/jenkinsci/jenkins/pull/4387[example])
+* It would make sense to backport the change to LTS, a Jira issue should exist and be labeled as `lts-candidate`
+
+**Step 2: Marking for merge**. Once the checklist is passed,
+a Core PR Reviewer or a Maintainer can mark the pull request for merge.
+
+* `ready-for-merge` label is set
+* An explicit comment is added to the pull request so that other repository watchers are notified.
+  Example: `Thanks to all contributors! We consider this change as ready to be merged towards the next weekly release. It may be merged after 24hours if there is no negative feedback`
```

Comment:
```suggestion
  Example: _Thanks to all contributors! We consider this change as ready to be merged towards the next weekly release. It may be merged after 24hours if there is no negative feedback_
```